test/models/CommonModel.spec.ts

```diff
@@ -267,6 +310,15 @@ describe('CommonModel', function() {
         doc1 = CommonModel.mergeCommonModels(doc1, doc2, doc);
         expect(doc1.enum).toEqual(doc2.enum);
       });
+      test('Should not contain duplicate values', function() {
+        const doc: Schema = { };
+        let doc1 = CommonModel.toCommonModel(doc);
+        let doc2 = CommonModel.toCommonModel(doc);
+        doc2.enum = ["string"];
+        doc1.enum = ["string"];
+        doc1 = CommonModel.mergeCommonModels(doc1, doc2, doc);
+        expect(doc1.enum).toEqual(["string"]);
+      });
```

Comment:
I am assuming this is a real use case right? 🤔

Reply:
This is a copy of my specification file, `allOf` was causing this issue.

```
        allOf:
          - $ref: "#/components/schemas/EventEnvelope"
          - type: object
            required: [eventName, event]
```
and
```
    EventEnvelope:
      $id: EventEnvelope
      description: "Envelope shared between messages"
      type: object
      required:
        - eventName
        - eventId
        - timestamp
        - workspaceId
        - event
```

EventEnvelope is generic in specification.

Reply:
@czlowiek488 take notice that this comment is for enums. Where I now remove duplicate values (as with required) 🙂

Reply:
You are right. What I`m thinking is that `required` field may need another test.

Reply:
@czlowiek488 do you mean in terms of `allOf`?

Reply:
yes

Reply:
Good point.

I revisited the tests we have about merging `allOf` and they all ensure that the function `mergeCommonModels` are called. So It implicitly should work with this fix 🙂